### ISSUE TITLE
Work on Wine-6.0 (NSPA) Stable release

### ISSUE THREAD

**Issue by nine7nine** [nine7nine/pkgbuilds_nspa#12]
- Rebase all patchwork on Wine-6.0

Now that we have a stable wine release, I will be creating a stable release.

The plan is to support both futex multiple wait / futex2 implementations in the linux kernel, followed by rewriting the Fsync code in Wine to support either implementation via an env variable... I have the futex2 patch for fsync, but i haven't taken a close look yet - although, i suspect it should be trivial to support both implementations.

**Comment by nine7nine** [comment]
just an fyi,

my development ceased for a while, due to some life / work issues. I became extremely busy at work and had a death of a loved one just before christmas. So i wasn't in any position to be working on this stuff.

anyway, having a stable wine version was something that i had been looking forward too, rather than chasing every development release. So in the mid-term this will be much better anyway. especially when one considers that Zeb is wanting to come up with a more viable / complete solution (over using futexes) with the NTSYNC proposal.

So while that is going on, I cn keep a stable NSPA build, based on a stable Wine build.

**Comment by nine7nine** [comment]
I should have this updates by the end of the weekend.

working on the kernel first, then I'll be attacking Wine-NSPA next.

**Comment by nine7nine** [comment]
So it appears that Fsync is broken (yet again), on Wine v6.0 (Staging) specifically. I have some options here, including just using a bit of patience. It's highly doubtful that I will have an updated build up in the next day or two, but regardless, i will get it sorted out.

This is all fine and not totally unexpected. I could build a package on pre v6.0 sources, possibly based on 5.13 (as Valve Proton is), which I know works. -- that's a definite option... Likewise, i suppose could do something like wine-tkg is doing, which is reverting a bunch of commits and fixing up code to keep fsync working (but 'm not inclined to do that, as it looks like a hassle)... or option #3; reach out to Zeb and possibly just wait until Fsync is fixed on v6.0?

In the meantime, I think I'll build v6.0 with Esync, while at the same time; I'll continue updating/porting and splitting some of my own patches. (maybe 35% left to do). This way I can at least test v6.0, while taking care of the bulk of adjusting my patchsets to Wine's internal changes (which the unix code split in upstream wine)...

If it's going to take too long to get v6.0 working with Fsync -- then in a handful of days, I will simply rebase on 5.13 (as that's low-hanging fruit / the simplest way forward).

**Comment by robbert-vdh** [comment]
Fsync works on all Wine versions >= 5.20 with Tk-Glitch's and Glorious Eggroll's Wine and Proton forks, perhaps they use their own patches?

**Comment by nine7nine** [comment]
Yeah, I got wine v6.0 (upstream) working with Esync/Fsync, last night. Not sure what's up with Wine-Staging, but I'll have another stab at it, later in the week when i have time.
